tokio-console/src/state/tasks.rs

```diff
@@ -321,14 +321,17 @@ impl Task {
     }
 
     pub(crate) fn busy(&self, since: SystemTime) -> Duration {
-        if let (Some(last_poll_started), None) =
-            (self.stats.last_poll_started, self.stats.last_poll_ended)
-        {
+        let busy_time_in_poll = |started| {
             // in this case the task is being polled at the moment
-            let current_time_in_poll = since.duration_since(last_poll_started).unwrap_or_default();
-            return self.stats.busy + current_time_in_poll;
+            let current_time_in_poll = since.duration_since(started).unwrap_or_default();
+            self.stats.busy + current_time_in_poll
+        };
+
+        match (self.stats.last_poll_started, self.stats.last_poll_ended) {
+            (Some(started), Some(ended)) if started > ended => busy_time_in_poll(started),
+            (Some(started), _) => busy_time_in_poll(started),
```

Comment:
fwiw, i think we _could_ combine these two match arms, since `Option<T>` also implements `PartialOrd` when `T: PartialOrd`, and `None` compares less than `Some`...so we could write
```suggestion
            (Some(started), ended) if Some(started) > ended => busy_time_in_poll(started),
```
but on the other hand, I think having two match arms might be more understandable, so I'm not sure if this change would actually be _better_...

Reply:
actually, thinking about it, this _would_ allow us to get rid of the `busy_time_in_poll` closure, so this whole function could be 
```rust
    pub(crate) fn busy(&self, since: SystemTime) -> Duration {
        if self.started > self.ended {
            debug_assert!(
                self.started.is_some(),
                "if `started` is greater than `ended`, it should always be `Some`",
            );
            if let Some(started) = started {
                // the task is currently being polled
                let current_time_in_poll = since.duration_since(started).unwrap_or_default();
                return self.stats.busy + current_time_in_poll;
            }
        }

        self.stats.busy
    }
```
or something...do you think that's substantially clearer?

Reply:
the `debug_assert!` could also just be an `expect`:
```rust
    pub(crate) fn busy(&self, since: SystemTime) -> Duration {
        if self.started > self.ended {
            // the task is currently being polled
            let started = self.started
                .expect("if `started` is greater than `ended`, it should always be `Some`");
            let current_time_in_poll = since.duration_since(started).unwrap_or_default();
            return self.stats.busy + current_time_in_poll;
        }

        self.stats.busy
    }
````

Reply:
What about this variant on your second option? It's a little weird to safely unwrap the started time and then use the Option, but it seemed a bit cleaner and doesn't require the `.expect` or the `debug_assert!`.

```rust
    pub(crate) fn busy(&self, since: SystemTime) -> Duration {
        if let Some(started) = self.stats.last_poll_started {
            if self.stats.last_poll_started > self.stats.last_poll_ended {
                // in this case the task is being polled at the moment
                let current_time_in_poll = since.duration_since(started).unwrap_or_default();
                return self.stats.busy + current_time_in_poll;
            }
        }
        self.stats.busy
    }
```

Reply:
ah, yeah, that's nicer. good call.